### ISSUE TITLE
Starcounter installation is corrupted when used from another system user

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Home#441]
Steps to reproduce:

1. Log in with a user A.
2. Install Starcounter.
3. Make sure that it's possible to create a new Starcounter Visual Studio project and build it.
4. Log out.
5. Log in with a user B.
6. Try to perform the step 3. **Visual Studio does not have Starcounter extension available.**
7. Try to open any Starcounter solution. **Visual Studio complains about unsupported project type.**
8. Try to start Starcounter using CMD: `staradmin start server`. **Fails with access denied.**

![image](https://user-images.githubusercontent.com/6435556/45152094-59a37a80-b1d0-11e8-9e9e-ccc622d4d21e.png)

```
C:\Users\UserB>staradmin start server
Starting server (starting server)                                                                                       
ScErrBadServerConfig (SCERR2115): Server configuration is invalid.
Version: 0.0.0.
Help page: https://docs.starcounter.io/?q=SCERR2115.

Type "star -h" or "staradmin -h" to see help

Stack trace:
   at Starcounter.Server.ServerServiceProcess.WaitUntilServerIsOnlineOrSignalExit[T](Func`2 checkExited, T context) in C:\tc_work\sc-19388\Level1\src\Server\Starcounter.Server\ServerServiceProcess.cs:line 211
   at Starcounter.CLI.StartServerCLICommand.RunWithinContext() in C:\tc_work\sc-19388\Level1\src\Server\Starcounter.CLI\StartServerCLICommand.cs:line 53
   at Starcounter.CLI.StartServerCLICommand.Run() in C:\tc_work\sc-19388\Level1\src\Server\Starcounter.CLI\StartServerCLICommand.cs:line 40
Access to the path 'C:\Users\UserA\Documents\Starcounter\Personal\Personal.server.config' is denied.
```

**Comment by ubbeK** [comment]
cc @iqbalskd

**Comment by iqbalskd** [comment]
Actually two issues here

1- When ` UserA`  installs starcounter with default settings the repository is being created under `C:\Users\UserA\Documents\Starcounter\` which is not accessible to ` UserB`  resulting starcounter server failure with ` Access Denied` error. Works fine if we change the default repository path to somewhere else.
2-  The ` Starcounter Visual Studio` extension is being installed for the current user only, which is not available for `UserB` so starcounter apps for the other users will fail to load.

**Comment by ubbeK** [comment]
> The Starcounter Visual Studio extension is being installed for the current user only, which is not available for UserB so starcounter apps for the other users will fail to load.

Is it possible to invoke the Starcounter VS Extension so it is installed for all users?

**Comment by iqbalskd** [comment]
>Is it possible to invoke the Starcounter VS Extension installation so it is installed for all users?
Yes, `VSIXInstaller.exe` takes an argument `/admin` which installs the extension for all users.  For that may be we need to include `VSIXInstaller` with starcounter and invoke it using command line like this:
```VSIXInstaller.exe /admin path_to_the_vsix.vsix```

**Comment by ubbeK** [comment]
How is it downloaded and installed today?

**Comment by iqbalskd** [comment]
> How is it downloaded and installed today?

Currently the code invokes `Starcounter.VS.vsix` executable from the Installation directory  which installs it for the current user only.

**Comment by ubbeK** [comment]
> Currently the code invokes Starcounter.VS.vsix executable from the Installation directory which installs it for the current user only.

And that vsix file is downloaded or packed within the installer?

So if we simply installer the VS extension through `VSIXInstaller.exe /admin path_to_the_vsix.vsix`, then we are all set for that particular issue, correct?

**Comment by ubbeK** [comment]
@iqbalskd Please do some googling in how we can located `VSIXInstaller.exe` (form me it is located at `C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VSIXInstaller.exe`). As the `*.vsix` files seem to be able to find all installed Visual Studio versions, should also be able to locate it.

I had to run `"C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\Tools\VsDevCmd.bat"` to be able to find `VSIXInstaller.exe`.

Maybe it uses `C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe`

**Comment by iqbalskd** [comment]
- Installing the `Starcounter.VisualStudio` for all users is done
- How should we approach this part?
> When UserA installs starcounter with default settings the repository is being created under C:\Users\UserA\Documents\Starcounter\ which is not accessible to UserB resulting starcounter server failure with Access Denied error. Works fine if we change the default repository path to somewhere else.

**Comment by ubbeK** [comment]
@miyconst 

So by default when installing Starcounter, the repository will be put at `%HOME%\Documents\Starcounter` which is not accessible by other users. Should we take any action regarding this?

**Comment by ubbeK** [comment]
As we decided not to change the default path of Starcounter repository in [PR meeting](https://github.com/Starcounter/Minutes/issues/190#issuecomment-424722755) as it is considered to be personal.

This issue has been implemented. Closing.

**Comment by linkdata** [comment]
Does this mean I can close my old issue? https://github.com/Starcounter/level1/issues/4070

**Comment by miyconst** [comment]
Yes. Closed.

**Comment by mmnosek** [comment]
This still doesn't work. I am not able to use Starcounter when logged in to another system user. I installed Starcounter in a path that both admin and non-admin can access, but I still have an error from the OP.

**Comment by ubbeK** [comment]
> This still doesn't work. I am not able to use Starcounter when logged in to another system user. I installed Starcounter in a path that both admin and non-admin can access, but I still have an error from the OP.

Complete discussion at: https://starcounter.slack.com/archives/C5Q4Y78F5/p1549437252257200

**Comment by iqbalskd** [comment]
To work with Starcounter Visual Studio extension, a user much have `StarcounterBin`  Environment Variable set for his/her account.

**Comment by iqbalskd** [comment]
Added a section in the troubleshooting doc:
https://github.com/Starcounter/AdminTrack/blob/master/DevelopmentInstructions/DeveloperHelp/troubleshooting.md#unable-to-create-or-load-starcounter-project-in-visual-studio

**Comment by bigwad** [comment]
I'd suggest keeping it open. Troubleshooting is good for untypical situations, while this one should work flawlessly out of the box.

**Comment by miyconst** [comment]
Closing as won't fix in Starcounter 2.4 and not applicable in Starcounter 3.0.
